### PR TITLE
Avoid precision loss in DocValueFormat.RAW#parseLong

### LINKED ISSUE #38692: Precision loss of include/exclude filter values in terms aggregation
**Elasticsearch version** (`bin/elasticsearch --version`):
6.4.2

**Plugins installed**: []

**JVM version** (`java -version`):
```
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-2~deb9u1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```

**OS version** (`uname -a` if on a Unix-like system):
```
Linux 4.18.0-2-amd64 #1 SMP Debian 4.18.10-2 (2018-10-07) x86_64 GNU/Linux
```

**Description of the problem including expected versus actual behavior**:
When trying to filter the values for which buckets are created when using terms aggregation on a long field, it's not possible to use values larger than 53 bits. When specifying one in a list of include values, no bucket is created most of the time, and when specifying one in a list of exclude values, the respective bucket is not excluded. I expect that for fields of the long type, the full 64 bit range works.

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. `PUT /test '{"mappings":{"_doc":{"properties":{"field1":{"type":"long"}}}}}'`
 2. `POST /test/_doc '{"field1":0}'`
 2. `POST /test/_doc '{"field1":123456}'`
 2. `POST /test/_doc '{"field1":4094779560956318341}'`
 3. `POST /test/_doc/_search '{"size":0,"aggs":{"by_field1":{"terms":{"field":"field1","include":[0,123456,4094779560956318341]}}}}` returns only buckets for 0 and 123456
 3. `POST /test/_doc/_search '{"size":0,"aggs":{"by_field1":{"terms":{"field":"field1","exclude":[0,123456,4094779560956318341]}}}}` returns only bucket for 4094779560956318341

**Provide logs (if relevant)**:

The reason is apparently here:
https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/IncludeExclude.java#L633-L642
and here:
https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/DocValueFormat.java#L118-L126

### PR DESCRIPTION
**Pull request by tjwilson90**
fixes #38692

### CODE REVIEW

**Comment by polyfractal** [comment]
@not-napoleon you might have a better handle on this area than me, would you mind reviewing? :heart:

File: server/src/test/java/org/elasticsearch/search/DocValueFormatTests.java [hunk 1]
```diff
@@ -176,8 +177,8 @@ public void testRawParse() {
         assertEquals(1d, DocValueFormat.RAW.parseDouble("1", randomBoolean(), null), 0d);
         assertEquals(.5, DocValueFormat.RAW.parseDouble("0.5", randomBoolean(), null), 0d);
         // not checking exception messages as they could depend on the JVM
-        expectThrows(IllegalArgumentException.class, () -> DocValueFormat.RAW.parseLong("", randomBoolean(), null));
-        expectThrows(IllegalArgumentException.class, () -> DocValueFormat.RAW.parseLong("abc", randomBoolean(), null));
+        expectThrows(IllegalArgumentException.class, () -> DocValueFormat.RAW.parseDouble("", randomBoolean(), null));
+        expectThrows(IllegalArgumentException.class, () -> DocValueFormat.RAW.parseDouble("abc", randomBoolean(), null));
```
Comment (not-napoleon) [inline server/src/test/java/org/elasticsearch/search/DocValueFormatTests.java]: Nice catch, thanks!

**Comment by not-napoleon** [comment]
@elasticmachine test this please

**Review by not-napoleon — APPROVED** [review]
This looks fine to me, but I do wonder if there was some reason we weren't doing this in the first place.  @jpountz I think you originally wrote this, do you know of some reason we wouldn't want to try parsing as a long first before falling back to rounding?

**Comment by jpountz** [comment]
@not-napoleon No good reason, this is a true bug. :)

**Comment by not-napoleon** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by tjwilson90** [comment]
The 6.8 branch is still maintained, right? Could this fix be backported to that branch as well please?

**Pull request merged**